src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -179,6 +183,91 @@ public void writeVariableExpr_scopedStaticFinalDecl() {
     assertThat(writerVisitor.write()).isEqualTo("public static final boolean x");
   }
 
+  @Test  
+  public void writeNormalBlockComment() throws Exception {
+    String content = "this is a test comment";
+    BlockComment blockComment = BlockComment.builder().setComment(content).build();
+    String expected = "/** this is a test comment */\n";
+    String formattedComment = blockComment.accept(writerVisitor);
+    assertThat(formattedComment).isEqualTo(expected);
+  }
+
+  @Test
+  public void writeNormalLineComment()throws Exception {
+    String content = "this is a test comment";
+    LineComment lineComment = LineComment.builder().setComment(content).build();
+    String expected = "// this is a test comment\n";
+    String formattedComment = lineComment.accept(writerVisitor);
+    assertThat(formattedComment).isEqualTo(expected);
+  }
+
+  @Test
+  public void writeLongLineComment() throws Exception {
+    String content =
+        "this is a long test comment with so many words, hello world, hello again, hello for 3 times, blah, blah!";
+    LineComment lineComment = LineComment.builder().setComment(content).build();
+    String expected =
+        "// this is a long test comment with so many words, hello world, hello again, hello for 3 times,\n// blah, blah!\n";
+    String formattedComment = lineComment.accept(writerVisitor);
+    assertThat(formattedComment).isEqualTo(expected);
+  }
+
+  @Test
+  public void writeNormalJavaDocComment() throws Exception {
+    String content = "this is a test comment";
+    String deprecatedText = "Use the {@link ArchivedBookName} class instead.";
+    String paramName = "shelfName";
+    String paramDescription =  "The name of the shelf where books are published to.";
+    String htmlP1 = "This class provides the ability to make remote calls to the backing service through method calls that map to API methods. Sample code to get started:";
+    String htmlP2 = "The surface of this class includes several types of Java methods for each of the API's methods:";
+    String sampleCode =
```

Comment:
Consider using the `TryCatch` class to build up this sample code.

Reply:
that's a great idea, done!